### PR TITLE
Fix password reset endopints:

### PR DESCRIPTION
**Pull request by bryanchriswhite**
* move log statement into proper scope
* rename `User#_pendingHashPass` to `User#pendingHashPass`
* add `User#pendingHashPass` to user model schema
* attempt some timing attack mitigation agains `PATCH /resets/:token` route
* add `rawbody` middleware to `PATCH /users/:id` route
* omit `#pendingHashPass` from `User#toObject`

### CODE REVIEW

File: lib/server/routes/users.js [hunk 1]
```diff
@@ -286,6 +286,24 @@ UsersRouter.prototype.createPasswordResetToken = function(req, res, next) {
 UsersRouter.prototype.confirmPasswordReset = function(req, res, next) {
   const User = this.storage.models.User;
 
+  /*
+   * Mitigate timing attack
+   *  attempt to make non-lookup-responses take similar
+   *  amounts of time as lookup-responses
+   */
+  const count = User.count();
```
Comment (bookchin) [inline lib/server/routes/users.js]: I believe this returns a cursor, so you need to pass a query object and callback to get the count. Check the mongoose docs.

File: lib/server/routes/users.js [hunk 2]
```diff
@@ -286,6 +286,24 @@ UsersRouter.prototype.createPasswordResetToken = function(req, res, next) {
...
+  const rand = Math.floor(Math.random() * count);
+  const nextPlusTime = (value) => {
+    User.findOne().skip(rand); // do nothing with this record
```
Comment (bookchin) [inline lib/server/routes/users.js]: Same here, returns a cursor.

**Comment by bryanchriswhite** [comment]
@coveralls slow ur roll man...

**Comment by bryanchriswhite** [comment]
[Bridge-gui PR #58](https://github.com/Storj/bridge-gui/pull/58) is dependent on this PR.

**Comment by bookchin** [comment]
Looks good, with one nitpick - can you not align the assignments? It looks pretty but now any time a new assignment is added, you have to realign everything else. 

Not critical obviously, just a style preference.

**Comment by bryanchriswhite** [comment]
@bookchin done

**Pull request merged**